v1/topdown/cache/cache.go

```diff
@@ -24,16 +24,36 @@ const (
 	defaultStaleEntryEvictionPeriodSeconds   = int64(0)   // never
 )
 
+var interQueryBuiltinValueCacheDefaultConfigs = map[string]*InterQueryBuiltinValueCacheConfig{}
+
+func getDefaultInterQueryBuiltinValueCacheConfig(name string) *InterQueryBuiltinValueCacheConfig {
+	return interQueryBuiltinValueCacheDefaultConfigs[name]
+}
+
+// RegisterDefaultInterQueryBuiltinValueCacheConfig registers a default configuration for the inter-query value cache;
+// used when none has been explicitly configured.
+// To disable a named cache when not configured, pass a nil config.
+func RegisterDefaultInterQueryBuiltinValueCacheConfig(name string, config *InterQueryBuiltinValueCacheConfig) {
+	interQueryBuiltinValueCacheDefaultConfigs[name] = config
+}
+
 // Config represents the configuration for the inter-query builtin cache.
 type Config struct {
 	InterQueryBuiltinCache      InterQueryBuiltinCacheConfig      `json:"inter_query_builtin_cache"`
 	InterQueryBuiltinValueCache InterQueryBuiltinValueCacheConfig `json:"inter_query_builtin_value_cache"`
 }
 
+// NamedValueCacheConfig represents the configuration of a named cache that built-in functions can utilize.
+// A default configuration to be used if not explicitly configured can be registered using RegisterDefaultInterQueryBuiltinValueCacheConfig.
+type NamedValueCacheConfig struct {
+	MaxNumEntries *int `json:"max_num_entries,omitempty"`
+}
+
 // InterQueryBuiltinValueCacheConfig represents the configuration of the inter-query value cache that built-in functions can utilize.
 // MaxNumEntries - max number of cache entries
 type InterQueryBuiltinValueCacheConfig struct {
-	MaxNumEntries *int `json:"max_num_entries,omitempty"`
+	MaxNumEntries     *int                                          `json:"max_num_entries,omitempty"`
+	NamedCacheConfigs map[string]*InterQueryBuiltinValueCacheConfig `json:"named,omitempty"`
```

Comment:
Should we use `NamedValueCacheConfig` instead of `InterQueryBuiltinValueCacheConfig`? Maybe it's this way due to `interQueryBuiltinValueCacheDefaultConfigs`?

Reply:
Sure, I can rename 👍

Reply:
Oh, I had just missed changing this to the correct type in a previous refactoring 😄